records/2022-Q4-activity-report.md

```diff
@@ -0,0 +1,420 @@
+- Maven cannot resolve sbt plugin artifacts.
+- Other resolution tools, like Coursier, must implement some sbt-specific hack to resolve them.
+- Some Maven proxies cannot check the validity of sbt plugins and they reject them.
+- The number of downloads of sbt plugins are unavailable on [oss.sonatype.org](https://oss.sonatype.org/).
+- [javadoc.io](https://javadoc.io/) cannot find the scaladoc of sbt plugins.
+- And more
+
+We implemented a solution that enables a smooth transition to valid Maven pattern of sbt plugins in [sbt#7096](https://github.com/sbt/sbt/pull/7096). This solution will be available in sbt 1.9.x, with bidirectional compatibility: sbt 1.9.x will publish the valid and legacy artifacts of sbt plugins, and will be able to resolve the valid or legacy artifacts of sbt plugins.
+
+### Language Improvements
+
+For Scala 3.
+
+#### Clause Interleaving
+
+As a reminder, last quarter, [SIP-47 - Clause Interleaving](https://github.com/scala/improvement-proposals/blob/main/content/clause-interleaving.md) proposed by the Center, got accepted for implementation.
+As described in the Q3 report, it provides a way to have type parameter bounds be dependent on value parameters.
+
+During this quarter, work has continued on bringing this SIP into the compiler as an experimental feature.
+
+Particular attention has been ported to the [pull request](https://github.com/lampepfl/dotty/pull/14019) and commit messages, to make them as easy to read and understand as possible.
+This way it will be usable as a guide when writing future SIP implementations.
+
+The code itself also follows this principle, bringing well documented precise changes that make the project more simple, not less.
+
+#### Case Classes that Can Evolve in a Binary-Compatible Way
+
+Case classes are popular to achieve domain modeling. However, they
+come with an important drawback: it is impossible to apply any evolution
+(adding/removing fields on a case class) in a backward binary compatible way.
+This forces library authors to manually re-implement the features of
+case classes that they need (typically, structural equality) at the price of
+more verbosity. Other solutions include tools generating code (e.g.
+contraband), or experimental macro annotations (e.g. data-class, scalameta).
+
+In Q4, we submitted a [proposal](https://github.com/scala/improvement-proposals/pull/50) that has been actively
+discussed by the Committee. Ultimately, we decided to withdraw the proposal in favor of documenting a workaround
+that supports our use case without changing the language.
+
+### TASTy Manipulation Library (tasty-query)
+
+For Scala 3.
+
+tasty-query is a library to read semantic information from Scala 3 classpaths.
+It reads Scala .tasty files, Scala 2 pickles and Java .class files, and presents the semantic information they contain in a unified API.
+
+tasty-query will be an essential tool for any static analysis involving Scala code.
+In particular, it is the basis for TASTy-MiMa.
```

Comment:
Can we link to TASTy-MiMa?

Reply:
I see we have a section down below that links to it, but maybe we should still link to the project or just the section down below.

Reply:
done in fce926e36f94706c5bedd331b59ffca522fecc59